### ISSUE TITLE
Wrong Date format for Visa Credit Card

### ISSUE THREAD

**Issue by mfernau** [hamvocke/dkb2homebank#19]
Hello hamvocke,

it seems that you use the wrong date format for the visa credit card import. 
I'm getting the error:
`ValueError: time data '21.10.23' does not match format '%d.%m.%Y'`

I changed the line 108 to the following code and it seems that the convert is successful then:
`'date': convert_short_date(row["wertstellung"]),`

**Comment by hamvocke** [comment]
Thanks for your bug reports (both, I saw the other one and will look into it in a moment).

It seems like the CSV formats over at DKB are currently in a weird state of change as they're moving more and more stuff over to their new web portal. It's a little tricky to stay ahead of these changes, so these bug reports are super helpful to me.

I assume you generated the CSV file using DKB's new web portal? If so, could you try to see if converting it using the `--giro` flag would work? Unfortunately, I don't have a Visa account type myself anymore, therefore I can't generate some test data for a Visa account myself. But it looks like they started using the same CSV format for all account types in the new banking portal.

It would be *super* helpful if you could help me out here with some test data. If you could

1. generate a Visa CSV file from the **new** banking portal
2. anonymize it (replace all the amounts and names with dummy values)
3. post it here

that would be incredibly helpful and help me better understand what the Visa CSV file looks like. I know there's a lot of sensitive information in those reports. I don't need a lot of data, just a file that contains 2 or 3 transactions (you can delete the rest) with some anonymized data would be perfect.

**Comment by mfernau** [comment]
I really like to help you out with a real Visa-Export example. But the DKB is currently more offline than online. I (again) can't login into my bank account...

But if I'm able to login again I'm going to export a fresh new CSV for you and post it here

**Comment by mfernau** [comment]
Finally I was able to login and export a new CSV-File for my VISA card.
As I'm not using my credit card very often, this file is not very "exciting". However - maybe it is helpful for your implementation..

Here you go
[visa.csv](https://github.com/hamvocke/dkb2homebank/files/13205232/visa.csv)

**Comment by hamvocke** [comment]
Thanks a ton! Turns out the Visa format generated by the new banking portal changed in quite a few ways indeed.

The latest commit should add the capability of dealing with visa CSV reports generated by the new banking portal.

Also a heads up: I removed the `--cash`, `--visa` and `--giro` flags from the script in favor of automatically detecting the file you're trying to convert (see release notes in the README.md). Simply invoking the script without these flags should do the trick.
